### ISSUE TITLE
Stop Resetting Filter When Switching Personalities

### ISSUE THREAD

**Issue by srguglielmo** [EDCD/EDDI#2436]
## What happens now
Prereq: Have multiple personalities (e.g. "personalityA" and "personalityB").
1. Go to the Speech Responder tab.
2. Select a personality (e.g. "personalityA").
3. Enter a word/phrase in the Filter input box (e.g. "interdiction").
4. Switch to another personality (e.g. "personalityB").
5. Notice the Filter is now cleared and all events are displayed.

## What I'd like to happen 
1. Go to the Speech Responder tab.
2. Select a personality (e.g. "personalityA").
3. Enter a word/phrase in the Filter input box (e.g. "interdiction").
4. Switch to another personality (e.g. "personalityB").
5. The Filter is kept and only the matching events are displayed.

tl;dr When switching personalities, don't reset the Filter. I have multiple personalities and frequently switch between them to compare as I'm editing/updating them. Right now, I either have to avoid using filter entirely (and just scroll to the event I want) or re-type my filter every time I switch. This is a very minor UX issue, but it would be nice to have.

## How it can happen
I'm not quite sure. There are several instances of `search_filter` in the code, but I don't see where something could be modified to save and load the filter when switching personalities.

## EDDI Version
4.0.2-b2
